presto-orc/src/main/java/com/facebook/presto/orc/reader/NullWrappingColumnReader.java

```diff
@@ -91,18 +92,12 @@ protected void makeInnerQualifyingSet()
 
     protected void addNullToKeep(int position, int inputIndex)
     {
-        if (nullsToAdd == null) {
-            nullsToAdd = new int[100];
-        }
-        else if (nullsToAdd.length <= numNullsToAdd) {
-            nullsToAdd = Arrays.copyOf(nullsToAdd, nullsToAdd.length * 2);
+        if (nullsToAdd == null || nullsToAdd.length <= numNullsToAdd) {
+            nullsToAdd = resize(nullsToAdd, numNullsToAdd);
         }
 
-        if (nullsToAddIndexes == null) {
-            nullsToAddIndexes = new int[nullsToAdd.length];
-        }
-        else if (nullsToAddIndexes.length < nullsToAdd.length) {
-            nullsToAddIndexes = Arrays.copyOf(nullsToAddIndexes, nullsToAdd.length);
+        if (nullsToAddIndexes == null || nullsToAddIndexes.length < nullsToAdd.length) {
```

Comment:
it would be simpler to resize both arrays at the same time

```
        if (nullsToAdd == null || nullsToAdd.length <= numNullsToAdd) {
            nullsToAdd = ensureCapacity(nullsToAdd, numNullsToAdd);
            nullsToAddIndexes = ensureCapacity(nullsToAddIndexes, numNullsToAdd);
        }
```